src/test/kotlin/snyk/common/CustomEndpointsTest.kt

```diff
@@ -94,4 +96,52 @@ class CustomEndpointsTest {
 
         assertThat(settingsUrl, equalTo("https://app.random-uuid.snyk.io/manage/snyk-code"))
     }
+
+    @Test
+    fun `isSnykAPI true when api subdomain and snyk domain`() {
+        val uri = URI("https://api.snyk.io")
+        assertTrue(uri.isSnykApi())
+    }
+
+    @Test
+    fun `isSnykAPI false for api subdomain and not snyk domain`() {
+        val uri = URI("https://api.NOTSNYK.io")
+        assertFalse(uri.isSnykApi())
+    }
+
+    @Test
+    fun `isSnykAPI false for non api path and snyk domain`() {
+        val uri = URI("https://snyk.io/notapi")
+        assertFalse(uri.isSnykApi())
+    }
+
+    @Test
+    fun `isSnykAPI true for api path and snyk domain`() {
+        val uri = URI("https://snyk.io/api")
+        assertTrue(uri.isSnykApi())
+    }
+
+    @Test
+    fun `non-api snyk domain does not need token`() {
+        val uri = "https://snyk.io"
+        assertFalse(needsSnykToken(uri))
+    }
+
+    @Test
+    fun `non snyk domain does not need token`() {
+        val uri = "https://NOTSNYK.io"
+        assertFalse(needsSnykToken(uri))
+    }
+
+    @Test
+    fun `api snyk path needs token`() {
+        val uri = "https://snyk.io/api"
+        assertTrue(needsSnykToken(uri))
+    }
+
+    @Test
+    fun `api snyk domain needs token`() {
```

Comment:
Suggestion: add tests for the officially documented endpoints as well eg. https://app.eu.snyk.io/api